opensearchpy/connection/http_async.py

```diff
@@ -65,10 +65,13 @@ def __init__(
 
         if http_auth is not None:
             if isinstance(http_auth, (tuple, list)):
-                http_auth = ":".join(http_auth)
+                http_auth = tuple(http_auth)
             elif isinstance(http_auth, string_types):
                 http_auth = tuple(http_auth.split(":", 1))
 
+            if isinstance(http_auth, tuple):
+                http_auth = aiohttp.BasicAuth(login=http_auth[0], password=http_auth[1])
```

Comment:
I think we are being too clever here. 

- We turn a tuple into a ... tuple. Seems unnecessary.
- We turn a list into a tuple.
- We turn a `string_types` into a tuple.
- Finally, we turn anything into `aiohttp.BasicAuth`.

Let's not be too clever, skip the intermediate `tuple` state and just make `aiohttp.BasicAuth` for any of the types?

```python
if isinstance(http_auth, tuple):
   http_auth = aiohttp.BasicAuth(login=http_auth[0], password=http_auth[1])
elif isinstance(http_auth, list)
   http_auth = aiohttp.BasicAuth(login=http_auth[0], password=http_auth[1])
elif isinstance(http_auth, string_types):
  http_auth = tuple(http_auth.split(":", 1))
```

Reply:
I tend to prefer DRY approaches, but that works too. I'll change it.

Reply:
Well, you choose what to DRY, `tuple` or `BasicAuth` :)

Reply:
Fair lol